packages/block-library/src/playlist/edit.js

```diff
@@ -0,0 +1,469 @@
+		dangerouslySetInnerHTML: {
+			__html: safeHTML( track?.title ? track.title : __( 'Untitled' ) ),
+		},
+	};
+	const trackArtist = {
+		dangerouslySetInnerHTML: {
+			__html: safeHTML(
+				track?.artist ? track.artist : __( 'Unknown artist' )
+			),
+		},
+	};
+	const trackAlbum = {
+		dangerouslySetInnerHTML: {
+			__html: safeHTML(
+				track?.album ? track.album : __( 'Unknown album' )
+			),
+		},
+	};
+
+	let ariaLabel;
+	if ( track?.title && track?.artist && track?.album ) {
+		ariaLabel = stripHTML(
+			sprintf(
+				/* translators: %1$s: track title, %2$s artist name, %3$s: album name. */
+				_x(
+					'%1$s by %2$s from the album %3$s',
+					'track title, artist name, album name'
+				),
+				track?.title,
+				track?.artist,
+				track?.album
+			)
+		);
+	} else if ( track?.title ) {
+		ariaLabel = stripHTML( track.title );
+	} else {
+		ariaLabel = stripHTML( __( 'Untitled' ) );
+	}
+
+	return (
+		<>
+			<div className="wp-block-playlist__current-item">
+				{ showImages && track?.image && (
+					<img
+						className="wp-block-playlist__item-image"
+						src={ track.image }
+						alt=""
+						width="70px"
+						height="70px"
+					/>
+				) }
+				<div>
+					{ ! track?.title ? (
+						<span className="wp-block-playlist__item-title">
+							<Spinner />
+						</span>
+					) : (
+						<span
+							className="wp-block-playlist__item-title"
+							{ ...trackTitle }
+						/>
+					) }
+					<div className="wp-block-playlist__current-item-artist-album">
+						<span
+							className="wp-block-playlist__item-artist"
+							{ ...trackArtist }
+						/>
+						<span
+							className="wp-block-playlist__item-album"
+							{ ...trackAlbum }
+						/>
+					</div>
+				</div>
+			</div>
+			<audio
+				controls="controls"
+				src={ track?.url ? track.url : '' }
+				onEnded={ onTrackEnd }
+				aria-label={ ariaLabel }
+				tabIndex={ 0 }
+			/>
+		</>
+	);
+};
+
+const PlaylistEdit = ( {
+	attributes,
+	setAttributes,
+	isSelected,
+	insertBlocksAfter,
+	clientId,
+} ) => {
+	const {
+		order,
+		showTracklist,
+		showNumbers,
+		showImages,
+		showArtists,
+		currentTrack,
+		tagName: TagName = showNumbers ? 'ol' : 'ul',
+	} = attributes;
+	const [ trackListIndex, setTrackListIndex ] = useState( 0 );
+	const blockProps = useBlockProps();
+	const { replaceInnerBlocks, __unstableMarkNextChangeAsNotPersistent } =
+		useDispatch( blockEditorStore );
+	const { createErrorNotice } = useDispatch( noticesStore );
+	function onUploadError( message ) {
+		createErrorNotice( message, { type: 'snackbar' } );
+	}
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const { innerBlockTracks } = useSelect(
+		( select ) => {
+			const { getBlock: _getBlock } = select( blockEditorStore );
+			return {
+				innerBlockTracks: _getBlock( clientId )?.innerBlocks ?? [],
+			};
+		},
+		[ clientId ]
+	);
+
+	// Ensure that each inner block has a unique ID,
+	// even if a track is duplicated.
+	useEffect( () => {
```

Comment:
I think this could be fixed in a follow-up, but it looks like this useEffect is running on every render. We could try memoising the unique IDs to avoid unnecessary re-renders.

Something like:

```
const uniqueIds = useMemo( () => {
	return innerBlockTracks.map( ( block ) => block.attributes.uniqueId );
}, [ innerBlockTracks ] );
```

And then add `uniqueIds` as a dependency to the useEffect.